go-controller/pkg/ovn/zone_interconnect/zone_ic_handler_test.go

```diff
@@ -399,6 +401,85 @@ var _ = ginkgo.Describe("Zone Interconnect Operations", func() {
+
+				dbSetup := libovsdbtest.TestSetup{
+					NBData: initialNBDB,
+					SBData: initialSBDB,
+				}
+
+				_, err := config.InitConfig(ctx, nil, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(config.IPv4Mode).To(gomega.BeTrue())
+				gomega.Expect(config.IPv6Mode).To(gomega.BeTrue())
+				config.Kubernetes.HostNetworkNamespace = ""
+
+				var libovsdbOvnNBClient, libovsdbOvnSBClient libovsdbclient.Client
+				libovsdbOvnNBClient, libovsdbOvnSBClient, libovsdbCleanup, err = libovsdbtest.NewNBSBTestHarness(dbSetup)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(libovsdbOvnSBClient.Connected()).To(gomega.BeTrue())
+				gomega.Expect(libovsdbOvnNBClient.Connected()).To(gomega.BeTrue())
+
+				err = createTransitSwitchPortBindings(libovsdbOvnSBClient, types.DefaultNetworkName, &testNode1, &testNode2, &testNode3)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				zoneICHandler := NewZoneInterconnectHandler(&util.DefaultNetInfo{}, libovsdbOvnNBClient, libovsdbOvnSBClient)
+				err = invokeICHandlerAddNodeFunction("global", zoneICHandler, &testNode1, &testNode2, &testNode3)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = checkInterconnectResources("global", types.DefaultNetworkName, libovsdbOvnNBClient, testNodesRouteInfo, &testNode1, &testNode2, &testNode3)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				// Set annotations to include ipv6 to node3 (remote zone)
+				node3Ipv6Subnet := "aef0:0:0:4::2895/64"
+				_, node3Ipv6SubnetNet, err := net.ParseCIDR(node3Ipv6Subnet)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				node3Ipv6SubnetPrefix := node3Ipv6SubnetNet.IP.String() + "/64"
+				node3TransitIpv6 := "fd97::4"
+
+				testNode3.Annotations[ovnNodeSubnetsAnnotation] = "{\"default\":[\"10.244.4.0/24\", \"" + node3Ipv6Subnet + "\"]}"
+				testNode3.Annotations[ovnTransitSwitchPortAddrAnnotation] = "{\"ipv4\":\"168.254.0.4/16\", \"ipv6\":\"" + node3TransitIpv6 + "/64\"}"
+
+				err = invokeICHandlerAddNodeFunction("global", zoneICHandler, &testNode3)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				r := nbdb.LogicalRouter{
+					Name: getNetworkScopedName(types.DefaultNetworkName, types.OVNClusterRouter),
+				}
+				clusterRouter, err := libovsdbops.GetLogicalRouter(libovsdbOvnNBClient, &r)
+				if err != nil {
+					return fmt.Errorf("could not find cluster router %s in the nb db for default network : err - %w", r.Name, err)
+				}
+
+				// Make sure cluster router now has an static route for the remote node's ipv6
+				staticRouteFound := false
+				for _, srUUID := range clusterRouter.StaticRoutes {
+					newPredicate := func(item *nbdb.LogicalRouterStaticRoute) bool {
+						return item.UUID == srUUID
+					}
+					sr, err := libovsdbops.FindLogicalRouterStaticRoutesWithPredicate(libovsdbOvnNBClient, newPredicate)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					gomega.Expect(sr).Should(gomega.HaveLen(1))
```

Comment:
hmm why only 1? why don't we find the v4 route as well since its a dualstack test?

Reply:
that is bc the loop walks each one of the static route from the logical router and looks it up by uuid. It should be getting 1 at each iteration.